src/category_theory/linear/linear_functor.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.additive_functor
+import category_theory.linear
+
+/-!
+# Linear Functors
+
+An additive functor between two `R`-linear categories is called *linear*
+if the induced map on hom types is a morphism of `R`-modules.
+
+# Implementation details
+
+`functor.linear` is a `Prop`-valued class, defined by saying that
+for every two objects `X` and `Y`, the map
+`F.map : (X ⟶ Y) → (F.obj X ⟶ F.obj Y)` is a morphism of `R`-modules.
+
+-/
+
+namespace category_theory
+
+variables (R : Type*) [ring R]
+
+/-- An additive functor `F` is `R`-linear provided `F.map` is an `R`-module morphism. -/
+class functor.linear {C D : Type*} [category C] [category D]
```

Comment:
Do sensible universe variables get picked up here? I guess I'm concerned that the universe variables to `category` might get mis-inferred as `max _ _` which can sometimes happen

Reply:
Yes, they are all independent, and no `max`s appear.

I'm a little torn here: we could spell out all the universes so no one needs to worry, or we could leave them all out because it seems to be working out of the box and is less cluttered.

For now I'm happy to leave them all out.